variation-commons-mongodb/src/test/java/uk/ac/ebi/eva/commons/mongodb/writers/VariantSourceMongoWriterTest.java

```diff
@@ -0,0 +1,182 @@
+        DBCollection fileCollection = mongoOperations.getCollection (collectionName);
+
+        VariantSourceMongoWriter filesWriter = new VariantSourceMongoWriter(
+                mongoOperations, collectionName);
+
+        VariantSourceMongo variantSource = getVariantSource();
+        filesWriter.write(Collections.singletonList(variantSource));
+
+        DBCursor cursor = fileCollection.find();
+        int count = 0;
+
+        while (cursor.hasNext()) {
+            count++;
+            DBObject next = cursor.next();
+            assertNotNull(next.get(VariantSourceMongo.FILEID_FIELD));
+            assertNotNull(next.get(VariantSourceMongo.FILENAME_FIELD));
+            assertNotNull(next.get(VariantSourceMongo.STUDYID_FIELD));
+            assertNotNull(next.get(VariantSourceMongo.STUDYNAME_FIELD));
+            assertNotNull(next.get(VariantSourceMongo.STUDYTYPE_FIELD));
+            assertNotNull(next.get(VariantSourceMongo.AGGREGATION_FIELD));
+            assertNotNull(next.get(VariantSourceMongo.SAMPLES_FIELD));
+            assertNotNull(next.get(VariantSourceMongo.DATE_FIELD));
+
+            DBObject meta = (DBObject) next.get(VariantSourceMongo.METADATA_FIELD);
+            assertNotNull(meta);
+            assertNotNull(meta.get("ALT"));
+            assertNotNull(meta.get("FILTER"));
+            assertNotNull(meta.get("INFO"));
+            assertNotNull(meta.get("FORMAT"));
+        }
+        assertEquals(1, count);
+    }
+
+    @Test
+    public void shouldWriteSamplesWithDotsInName() throws Exception {
+
+        String collectionName = COLLECTION_FILES_NAME + ThreadLocalRandom.current().nextInt(0, 100000);
+        DBCollection fileCollection = mongoOperations.getCollection(collectionName);
+
+        VariantSourceMongoWriter filesWriter = new VariantSourceMongoWriter(
+                mongoOperations, collectionName);
+
+        VariantSourceMongo variantSource = getVariantSource();
+        Map<String, Integer> samplesPosition = new HashMap<>();
+        samplesPosition.put("EUnothing", 1);
+        samplesPosition.put("NA.dot", 2);
+        samplesPosition.put("JP-dash", 3);
+        variantSource.setSamplesPosition(samplesPosition);
+
+        filesWriter.write(Collections.singletonList(variantSource));
+
+        DBCursor cursor = fileCollection.find();
+
+        while (cursor.hasNext()) {
+            DBObject next = cursor.next();
+            DBObject samples = (DBObject) next.get(VariantSourceMongo.SAMPLES_FIELD);
+            Set<String> keySet = samples.keySet();
+
+            Set<String> expectedKeySet = new TreeSet<>(Arrays.asList("EUnothing", "NA£dot", "JP-dash"));
+            assertEquals(expectedKeySet, keySet);
+        }
+    }
+
+    @Test
+    public void shouldCreateUniqueFileIndex() throws Exception {
+
+        String collectionName = COLLECTION_FILES_NAME + ThreadLocalRandom.current().nextInt(0, 100000);
```

Comment:
I'm more or less ok with this, given that the probability is sum_i=1^n{i}/100000, which means that making 10 builds per day, in 90 years there will be 1 run that will fail. But if we add 7 tests to this class, it will take 6 months at 10 builds per day.

The point I'm making is that with a setup and teardown methods (like VariantMongoWriterTest) there's no need to introduce randomness.

The risk is calculated, but man, I'm bad at maths.